incubator/virtualcluster/README.md

```diff
@@ -35,36 +35,78 @@ Kubecon EU 2020 talk (~25 mins) | WG meeting demo (~50 mins)
+In VirtualCluster, tenant master owns the source of the truth for the specs of all the synced objects. 
+The exceptions are persistence volume and storage class objects whose source of the truth is the super cluster.
+The syncer along with the super cluster acts like a resource provider controller which updates the synced
+object's status in each tenant master. This abstraction model means the following assumptions:
+- The synced object spec should not be altered by any arbitrary controller in the super cluster.
+- Tenant master owns the life cycle management for the synced object. The synced objects should not be
+  managed by any controllers (say StatefulSet for example) in the super cluster.
+
+If any of the above assumptions is violated, VirtualCluster may not work as expected. Note that this 
+does not mean that a cluster administrator cannot install webhooks, for example, a sidecar webhook, 
+in the super cluster. Those webhooks still work but the cluster administrator admits that the changes are going
+to be hidden to tenants. Alternatively, those webhooks can be installed in tenant masters so that
+tenants will be aware of all changes.
+
+## Limitations
+
+Ideally, tenants should not be aware of the existence of the super cluster in most cases. 
+There are still some noticeable differences comparing a tenant master and a normal kubernetes cluster.
+
+- In tenant master, node objects only show up after tenant Pods are created. The super cluster
+  node topology is not fully exposed in the tenant master. This means VirtualCluster does not support
+  `DaemonSet` alike workloads in tenant master. Currently, the syncer controller rejects a newly
+  created tenant Pod if its `nodename` has been set in the spec. 
 
-**Here are other considerations that users should be aware of:**
+- The syncer controller manages the life cycle of the node objects in tenant master but
+  it does not update the node lease objects in order to reduce network traffics. As a result,
+  it is recommended to increase the tenant master node controller `--node-monitor-grace-period` 
+  parameter to a larger value ( >60 seconds, done in the sample clusterversion
+  [yaml](config/sampleswithspec/clusterversion_v1_nodeport.yaml) already).
+
+- Coredns is not tenant-aware. Hence, tenants should install coredns in tenant masters if DNS is required.
+The DNS service should be created in the `kube-system` namespace using the name `kube-dns`. The syncer controller can then
+recognize the DNS service cluster IP in super cluster and inject it into Pod spec `dnsConfig`.
+
+- The cluster IP field in the tenant Service Spec is a bogus value. If any tenant controller requires the
+actual cluster IP that takes effect in the super cluster nodes, a special handling is required. 
+The workaround usually is going to be a simple code change in the controller. This 
+[document](./doc/tenant-dns.md) shows an example for coredns.
```

Comment:
I wonder if in this section we can write out the annotation so that someone could look up the clusterIP without having to read the tenant-dns doc?